ui/app/models/pki/pki-role-engine.js

```diff
@@ -77,6 +78,147 @@ export default class PkiRoleEngineModel extends Model {
     editType: 'boolean',
   })
   addBasicConstraints;
+  /* End of overriding default options */
+
+  /* Overriding Domain handling options */
+  @attr({
+    label: 'Allowed Domains',
+    subText: 'Specifies the domains this role is allowed to issue certificates for. Add one item per row.',
+    editType: 'stringArray',
+  })
+  allowedDomains;
+
+  @attr('boolean', {
+    label: 'Allow templates in allowed domains',
+  })
+  allowedDomainsTemplate;
+
+  @attr('string', {
+    editType: 'moreInfo',
+    text: 'These options can interact intricately with one another. For more information,',
+    docText: 'learn more here.',
+    docLink: '/api-docs/secret/pki#allowed_domains',
+  })
+  moreInfo;
+  /* End of overriding Domain handling options */
+
+  /* Overriding Key parameters options */
+  @attr('string', {
+    subText: `These are the parameters for generating or validating the certificate's key material.`,
+    dontShowInput: true,
+    dontShowLabel: true,
+  })
+  keyParametersLabel;
+
+  @attr('string', {
+    label: 'Key type',
+    possibleValues: ['rsa', 'ec', 'ed25519', 'any'],
+    defaultValue: 'rsa',
+  })
+  keyType;
+
+  @attr('string', {
+    label: 'Key bits',
+  })
+  keyBits;
+
+  // "possibleValues" for the field "keyBits" depends on the value of the selected "keyType"
+  get keyBitsConditional() {
+    const keyBitOptions = {
+      rsa: [2048, 3072, 4096],
+      ec: [256, 224, 384, 521],
+      ed25519: [0],
+      any: [0],
+    };
+    const attrs = expandAttributeMeta(this, ['keyBits']);
+    attrs[0].options['possibleValues'] = keyBitOptions[this.keyType];
+    return attrs[0];
+  }
+
+  @attr('number', {
+    label: 'Signature bits',
+    subText: `Only applicable for key_type 'RSA'. Ignore for other key types.`,
+    defaultValue: 0,
+    possibleValues: [
+      {
+        value: 0,
+        displayName: 'Defaults to 0',
+      },
+      {
+        value: 256,
+        displayName: '256 for SHA-2-256',
+      },
+      {
+        value: 384,
+        displayName: '384 for SHA-2-384',
+      },
+      {
+        value: 512,
+        displayName: '512 for SHA-2-5124',
+      },
+    ],
+  })
+  signatureBits;
+  /* End of overriding Key parameters options */
+
+  /* Overriding Key usage options */
+  // The following turns options that go into an arrayList for param "key_usage" and "ext_key_usage" into checkboxes
+  @attr('string', {
+    label: 'Key usage',
+    subText: `Specifies the default key usage constraint on the issued certificate. To specify no default key_usage constraints, uncheck every item in this list.`,
+    dontShowInput: true,
+    isTitleOfGridGroup: true,
+  })
+  keyUsageLabel;
+
+  // https://pkg.go.dev/crypto/x509#KeyUsage
+  @attr('boolean', {
+    defaultValue: true,
+  })
+  DigitalSignature;
```

Comment:
Following up from our conversation and since these properties don't come from the server and aren't sent as properties in POST/PUT I think it's a good idea to remove these and rework how the key usage check boxes function.